### ISSUE TITLE
Upload hangs on Thunderbird 52

### ISSUE THREAD

**Issue by okmikel** [wavexx/dl#42]
Uploading a file with the addon in Thunderbird 52 hangs with a spinning circle in Windows 10.

This is shown in the console of Thunderbird:

> TypeError: Not enough arguments to File.[Weitere Informationen]  nsDL.js:108:36
> ReferenceError: listener is not defined[Weitere Informationen]  MsgComposeCommands.js:1592:7

```
    if(!msg && !file)
      req.send();
    else
    {
      let data = Cc["@mozilla.org/files/formdata;1"].createInstance(Ci.nsIDOMFormData);
      if(msg) data.append("msg", JSON.stringify(msg));
      if(file) data.append("file", new File(file));
      req.send(data);
    }
```

**Comment by wavexx** [comment]
Which version of the addon are you using?
This should be fixed in DL 0.17.1

**Comment by okmikel** [comment]
I am using 0.17.1 and it worked before upgrading to 52.0 from 48.8. 
I did an update from Firefox 48.8 to 52.0. 
The problem with the hanging upload was also when updating to 48.0 a year ago as I remember and this was fixed with 0.17.1 I think.

Should I try to deinstall and reinstall 0.17.1 tomorrow?

**Comment by wavexx** [comment]
Uhm no, by reading the error again I see that the constructor for File objects changed so some tweaks are, sadly, necessary.

**Comment by wavexx** [comment]
On Wed, Apr 05 2017, okmikel wrote:
> I did an update from Firefox 48.8 to 52.0.

Could you please send me an email privately so I can send you some
extensions to try?

**Comment by wavexx** [comment]
Fixed in commit 1bbad3503d7274c8b0645cbd17474e322f895be5
The 0.17.2 is pending review on AMO, but I've already put it live on https://www.thregr.org/~wavexx/software/dl/thunderbird.html
